lib/magma.gd

```diff
@@ -562,6 +562,38 @@ InstallFactorMaintenance( IsCommutative,
 InstallTrueMethod( IsCommutative, IsMagma and IsTrivial );
 
 
+#############################################################################
+##
+#P  IsFinitelyGeneratedMagma( <M> ) . . . . test whether a magma is fin. gen.
+##
+##  <#GAPDoc Label="IsFinitelyGeneratedMagma">
+##  <ManSection>
+##  <Prop Name="IsFinitelyGeneratedMagma" Arg='M'/>
+##
+##  <Description>
+##  A magma <A>M</A> is <E>finitely generated</E> if there is a finite subset
+##  <M>X</M> of the magma such that every element of the magma can be written
+##  as a product of the elements of <M>X</M>.
+##  <P/>
+##  Note that this is a pure existence statement. Even if a magma is known to
+##  be generated by a finite number of elements, it can be very hard or even
+##  impossible to obtain such a generating set if it is not known.
+##  <P/>
+##  Also note that the notion of being finitely generated is independent of
+##  whether the magma is considered as a magma, a magma-with-one or a
+##  magma-with-inverses.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareProperty( "IsFinitelyGeneratedMagma", IsMagma );
+
+InstallFactorMaintenance( IsFinitelyGeneratedMagma,
+    IsMagma and IsFinitelyGeneratedMagma, IsObject, IsMagma );
+
+InstallTrueMethod( IsFinitelyGeneratedMagma, IsMagma and IsFinite );
+
```

Comment:
Why not also `InstallTrueMethod( IsMagma, IsFinitelyGeneratedMagma );`, as for `IsFinitelyGeneratedGroup` and `IsFinitelyGeneratedMonoid`?

Reply:
Thanks, added